### DISCUSSION TITLE
[Docs-Tex.SetNativeSurface] (Your question)

### DISCUSSION THREAD

**Discussion by JackFang** [StereoKit/StereoKit#1096] [General]
How to use this API to implement an Android Surface display? could provide an example?

**Comment by maluoi** [comment]
I unfortunately don't have a direct sample of using an Android Surface in this context. But here's some code that consumes AR Core's surface using SetNativeSurface: https://github.com/maluoi/SKARCore/blob/main/ARCore.cs

A related topic for Android Surfaces is using composition layers for more performant and higher quality display of Android Surfaces in some contexts. This code is a helper for creating swapchains from Android Surfaces and submitting composition layers that use them: https://gist.github.com/maluoi/bc51eec1a56ec4bf7334d104cc8c630d
